Contest_API.md

```diff
@@ -114,36 +114,58 @@ their expected behavior, if implemented.
-    not need to be specified (other than in the URL) and if specified
-    different from in the URL, a `409 Conflict` HTTP code should be
-    returned.
+    Creates or replaces a specific element. This method is idempotent, can only
+    be called on a specific element, and replaces its contents with the
+    data provided. The payload data must be complete, i.e. the `id` is
+    required and no partial updates are allowed.
+
   - `PATCH`
-    Updates/modifies a specific element. Similar to `PUT` but allows
-    partial updates by providing only that data, for example:
-    `PATCH  https://example.com/api/contests/wf14/teams/10`
-    with JSON contents
-    `{"name":"Our cool new team name"}`
-    No updates of the `id` attribute are allowed either.
+    Updates/modifies a specific element. This method is idempotent,
+    can only be called on a specific element, and replaces the given
+    attributes with the data provided. For example
+    `PATCH https://example.com/api/contests/wf14/teams/10`
+    with JSON contents `{"name":"Our cool new team name"}`.
+
   - `DELETE`
-    Delete a specific element. Idempotent, but may return a 404 status
-    code when repeated. Any provided data is ignored. Example:
-    `DELETE  https://example.com/api/contests/wf14/teams/8`
-    Note that deletes must keep [referential
-    integrity](#referential-integrity) intact.
+    Delete a specific element. Idempotent, but will return a `404` error
+    code when repeated. Any provided data is ignored and there is no response body.
+    Example: `DELETE https://example.com/api/contests/wf14/teams/8`.
+    Note that deletes must keep [referential integrity](#referential-integrity) intact.
+
+#### Success, Failure, and HTTP Responses
+
+Standard [HTTP status codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes) are
+returned to indicate success or failure. If successful the response body of every method
+(except for DELETE, and GET on a collection) will contain the current (updated) state of the element.
+
+If a POST, PUT, PATCH would cause any of the following issues it will fail, in addition to any endpoint or element-specific requirements:
+
+* A PATCH on an `id` that doesn't exist. Will return a 404 error code.
+* A PUT or PATCH containing an id that does not match the URL. Will return a 409 error code.
+* A missing required attribute.
+* An attribute that must not be provided is provided.
+* An attribute type that is incorrect or otherwise invalid (e.g. non-nullable attribute set to null).
+* A reference to another element is invalid (to maintain [referential integrity](#referential-integrity)).
+
+A DELETE will fail in the following conditions, in addition to any endpoint or element-specific requirements:
+
+* Deleting an element `id` that doesn't exist. Will return a 404 error code.
+* If the element being deleted is referenced by another element (to maintain [referential integrity](#referential-integrity)).
```

Comment:
I think it should be allowed for the server to implement cascade instead. I.e., it should not be a requirement for this `DELETE` to fail.

Reply:
Fixed in new commit.